### DISCUSSION TITLE
Dear Jan , Could you please help me with the error of nwxtregress

### DISCUSSION THREAD

**Discussion by zhangzhqnk** [JanDitzen/nwxtregress#5] [Q&A]
Dear Jan, 
 I  run the demo code in your github  , eg VA data set to estimate a SAR mode, the estimation process goes smoothly but while I try to get the imact, estat impact, it returns the following errors:
 estat impact
----+--- 10 --+--- 20 --+--- 30 --+--- 40 --+--- 50   %
                       *:  3200  conformability error
        SparseMultiply():     -  function returned error
             BarryPace():     -  function returned error
        calc_DIEffects():     -  function returned error
                 <istmt>:     -  function returned error

As you methioned in one of the discussion, I  need to install the new version of moremata library in Stata. The problem is that even I install it, the errors are still there, I sincerely hope that you can help me.

the coding is just like this :

use IO
keep if Year == 1998
drop buyer_industry seller_industry
replace sam = 0 if sam < 0
replace sam = 0 if ID1==ID2
reshape wide sam, i(ID1) j(ID2)
spset ID1
spmatrix fromdata WSpmat = sam* , replace
clear
use VA
nwxtregress cap_cons compensation net_surplus , dvarlag(WSpmat) seed(1234)
estat impact

Thank you very much!

**Comment by JanDitzen** [comment]
Thanks for the pointer - this was indeed a bug in the code. The error occurred if a non sparse weight matrix was used.

I pushed an update to GitHub, please check and let me know if the error remains.

**Reply by zhangzhqnk** [reply]
Thanks a lot, Jan! It works. Terrific coding for spatial network estimation!

**Reply by JanDitzen** [reply]
Perfect, thanks a lot for letting me know. Please get in touch if you have any other questions or comments.

**Reply by dalv1990** [reply]
Dear Jan,
The nwxtregress is terrific! If I would like to refer to your related article, could you be willing to provide it?
Thank you very much!
